src/platform_impl/macos/window.rs

```diff
@@ -1233,6 +1240,21 @@ impl WindowExtMacOS for UnownedWindow {
         .setHasShadow_(if has_shadow { YES } else { NO })
     }
   }
+
+  #[inline]
+  fn set_titlebar_transparent(&self, transparent: bool) {
+    unsafe {
+      let id = self.ns_window() as cocoa::base::id;
+      let mut style_mask = id.styleMask();
+      style_mask.set(
+        NSWindowStyleMask::NSFullSizeContentViewWindowMask,
+        transparent,
+      );
+      self.ns_window.setStyleMask_(style_mask);
```

Comment:
This shouldn't be the responsibility of this method, it should probably be split into another method `WindowExtMacOS::set_fullsize_content_view`